Formula/lua-language-server.rb

```diff
@@ -0,0 +1,110 @@
+class LuaLanguageServer < Formula
+  desc "Language Server for the Lua language"
+  homepage "https://github.com/sumneko/lua-language-server"
+  # pull from git tag to get submodules
+  url "https://github.com/sumneko/lua-language-server.git",
+      tag:      "2.5.2",
+      revision: "cb2042160865589b5534a6bf0b6c366ae4ab1d99"
+  license "MIT"
+  head "https://github.com/sumneko/lua-language-server.git", branch: "master"
+
+  depends_on "ninja" => :build
+
+  on_linux do
+    depends_on "gcc" # For C++17
+  end
+
+  fails_with gcc: 5
+
+  # Disable `filesystem.test_appdata_path`.
+  #
+  # This test expeccts the application to run under the user's ${HOME},
+  # which is not compatible with homebrew's build environment.
+  # See https://github.com/actboy168/bee.lua/issues/21
+  # Adapted from https://github.com/macports/macports-ports/tree/master/lua/lua-language-server
+  patch :DATA
```

Comment:
Can we ask upstream if they'd consider using an option to disable this for installations outside `$HOME`?

Reply:
Macports already tried and got a hard "No"... :(

(See the linked issue.)

Reply:
Hmm. Ok. Have you tried doing
```ruby
mkdir_p "#{ENV["HOME"]}/Library/Caches"
```
?

What does `fs.appdata_path()` evaluate to when this test is run?

Reply:
```
3rd/bee.lua/test/test_filesystem.lua:38: expected: /private/tmp/lua-language-server-20211205-99244-tg7zbr/.brew_home/Library/Caches, actual: /Users/clason/Library/Caches
```

Reply:
Does it try to write to `/Users/clason/Library/Caches` during the build? I wonder if we can just do
```ruby
ENV.delete "HOME"
```

Reply:
So creating directories in `$HOME` won't help since the function that is tested does _not_ use `$HOME` -- for whatever reason...

Reply:
No, it just tests whether a function (that uses system libraries on macOS) returns the same as `$HOME/Library/Caches`.

Reply:
Ok, I've tried asking upstream again. Couldn't hurt, I guess. Let's see what they say.

Reply:
Upstream just (sulkily) removed the test for macOS :/

I'm trying the new version without the local modification to see what happens on Linux, but I suspect it is still required.